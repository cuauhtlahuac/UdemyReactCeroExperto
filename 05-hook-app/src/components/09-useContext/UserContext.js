import { createContext } from "react";

// Creamos un High Order Component
export const UserContext = createContext(null);