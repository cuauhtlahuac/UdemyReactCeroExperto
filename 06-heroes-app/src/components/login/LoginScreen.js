import React, { useContext } from 'react';
import { AuthContext } from 'contexts/authContext';
import { types } from 'types/types';

const LoginScreen = ({ history }) => {
	const { dispatch } = useContext(AuthContext);

	const handleLogin = () => {
		dispatch({
			type: types.login,
			payload: { name: 'Cuauhtlahuac' },
		});

		history.replace('/');
	};

	return (
		<div className="m-5">
			<h1>Login</h1>
			<hr />

			<button className="btn btn-primary" onClick={handleLogin}>
				Login
			</button>
		</div>
	);
};

export default LoginScreen;
