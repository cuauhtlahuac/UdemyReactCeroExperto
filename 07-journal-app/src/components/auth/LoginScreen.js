import React from 'react'

const LoginScreen = () => {
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
    </div>
  )
}

export default LoginScreen
