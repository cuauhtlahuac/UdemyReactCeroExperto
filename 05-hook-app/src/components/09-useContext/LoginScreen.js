import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
	const { setUser } = useContext(UserContext);
	return (
		<div>
			<h1>LoginScreen</h1>
			<hr />
			<button
				className="btn btn-primary"
				onClick={() => setUser({ id: 132, name: 'Petronila' })}
			>
				Click
			</button>
		</div>
	);
};

export default LoginScreen;
