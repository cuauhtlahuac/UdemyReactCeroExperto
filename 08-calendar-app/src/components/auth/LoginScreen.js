import { useForm } from 'hooks/useForm';

import './login.css';

const LoginScreen = () => {
	const initialLoginState = {
		loginEmail: '',
		loginPassword: '',
	};

	const [ formLoginValues, handleLoginInputChange, reset ] = useForm(
		initialLoginState,
	);

	const { loginEmail, loginPassword } = formLoginValues;

	const handleLogin = event => {
		event.preventDefault();
		console.log(formLoginValues);
	};

	return (
		<div className="container login-container">
			<div className="row">
				<div className="col-md-6 login-form-1">
					<h3>Ingreso</h3>

					<form onSubmit={handleLogin}>
						<div className="form-group">
							<input
								type="text"
								className="form-control"
								placeholder="email"
								name="loginEmail"
								value={loginEmail}
								onChange={handleLoginInputChange}
							/>
						</div>
						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="password"
								name="loginPassword"
								value={loginPassword}
								onChange={handleLoginInputChange}
							/>
						</div>
						<div className="form-group">
							<input type="submit" className="btnSubmit" value="Login" />
						</div>
					</form>
				</div>

				<div className="col-md-6 login-form-2">
					<h3>Registro</h3>
					<form>
						<div className="form-group">
							<input
								type="text"
								className="form-control"
								placeholder="Nombre"
							/>
						</div>
						<div className="form-group">
							<input
								type="email"
								className="form-control"
								placeholder="Correo"
							/>
						</div>
						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="Contraseña"
							/>
						</div>

						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="Repita la contraseña"
							/>
						</div>

						<div className="form-group">
							<input type="submit" className="btnSubmit" value="Crear cuenta" />
						</div>
					</form>
				</div>
			</div>
		</div>
	);
};

export default LoginScreen;
