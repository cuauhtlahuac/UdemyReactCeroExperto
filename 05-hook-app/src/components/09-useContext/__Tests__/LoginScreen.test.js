import React from 'react';
import { mount } from 'enzyme';

import { UserContext } from '../UserContext';
import LoginScreen from '../LoginScreen';

describe('HomeScreen Tests', () => {
	const user = {
		name: 'Cuautlahuac',
		email: 'cuauh@gmail.com',
	};

	const component = mount(
		<UserContext.Provider value={user}>
			<LoginScreen />
		</UserContext.Provider>,
	);

	test('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
