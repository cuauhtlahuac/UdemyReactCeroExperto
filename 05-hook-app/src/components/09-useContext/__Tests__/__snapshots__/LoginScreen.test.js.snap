// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Tests should render correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;
